vaadin-grid-flow-integration-tests/benchmark/run-benchmark.js

```diff
@@ -0,0 +1,195 @@
+const { spawn, execSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+const GRID_DIR = '../../';
+const REF_GRID_DIR = './reference-clone';
+const TEST_DIR = 'vaadin-grid-flow-integration-tests';
+
+const gridPath = path.resolve(GRID_DIR);
+const refGridPath = path.resolve(REF_GRID_DIR);
+
+const gridTestPath = path.resolve(gridPath, TEST_DIR);
+const refGridTestPath = path.resolve(refGridPath, TEST_DIR);
+
+const resultsPath = path.resolve('./results');
+
+const REF_JETTY_PORT = 8088;
+// TODO: Fix as a Grid tag of a revision that will be used as the baseline/reference for the tests
+const REF_GIT_TAG = 'benchmark';
+
+const processes = [];
+const cleanup = () => processes.forEach((ps) => ps.kill());
+process.on('exit', cleanup);
+process.on('SIGINT', cleanup);
+
+const testVariants = [];
+['firefox-headless', 'chrome-headless'].forEach((browserName) => {
+  [
+    'simple',
+    'multicolumn',
+    'componentrenderers',
+    'detailsopened',
+    'tree',
+    'mixed',
+  ].forEach((gridVariantName) => {
+    testVariants.push({
+      gridVariantName,
+      browserName,
+      metricName: 'rendertime',
+    });
+    testVariants.push({
+      gridVariantName,
+      browserName,
+      metricName: 'verticalscrollframetime',
+    });
+    if (['tree', 'mixed'].includes(gridVariantName)) {
+      testVariants.push({
+        gridVariantName,
+        browserName,
+        metricName: 'expandtime',
+      });
+    }
+    if (['multicolumn', 'mixed'].includes(gridVariantName)) {
+      testVariants.push({
+        gridVariantName,
+        browserName,
+        metricName: 'horizontalscrollframetime',
+      });
+    }
+  });
+});
+
+const startJetty = (cwd) => {
+  return new Promise((resolve) => {
+    const jetty = spawn('mvn', ['jetty:run'], { cwd });
+    processes.push(jetty);
+    jetty.stderr.on('data', (data) => console.error(data.toString()));
+    jetty.stdout.on('data', (data) => {
+      console.log(data.toString());
+
+      if (data.toString().includes('Started Jetty Server')) {
+        resolve();
+      }
+    });
+  });
+};
+
+const prepareReferenceGrid = () => {
+  execSync(
+    `git clone --depth=1 --single-branch --branch ${REF_GIT_TAG} https://github.com/vaadin/vaadin-grid-flow.git ${refGridPath}`
+  );
+
+  // Add Jetty config to start the server on a different port
+  const pomFile = path.resolve(refGridTestPath, 'pom.xml');
+  const pomFileContent = fs.readFileSync(pomFile, 'utf8');
+
+  const result = pomFileContent.replace(
+    /<artifactId>jetty-maven-plugin<\/artifactId>/g,
+    `
+    <artifactId>jetty-maven-plugin</artifactId>
+      <configuration>
+        <httpConnector>
+          <port>${REF_JETTY_PORT}</port>
+        </httpConnector>
+        <stopPort>${REF_JETTY_PORT + 1}</stopPort>
+      </configuration>
+    `
+  );
```

Comment:
- This seems a bit fragile. Do you think this PR makes sense? https://github.com/vaadin/flow-component-base/pull/179
Then we could just run `mvn jetty:run -Djetty.http.port=SOMEPORT -Djetty.stop.port=SOMEOTHERPORT`

- Should we be more explicit about the stop port? Might make debugging easier in the case jetty fails to run because of port conflicts.

Reply:
Yes, would be more solid that way. Then the default 9998 port could also be enforced